plugins/deploy_waitlist/lib/samson_deploy_waitlist/waitlist_monitor.rb

```diff
@@ -0,0 +1,23 @@
+module SamsonDeployWaitlist
+  class WaitlistMonitor
+
+    TIME_TO_LOCK_QUEUE = 10
+
+    def self.check_your_head
+      Stage.all.each do |s|
+        waitlist = Waitlist.new(s.project_id, s.id)
+        next unless waitlist.head_updated_at.present? && waitlist.list.present?
+
+        current_head = waitlist.list[0]
+
+        next if s.lock.try(:user).try(:email) == current_head[:email]
```

Comment:
this is already bound to ruby > 2.4. You can use a safe operator
`next if s.lock&.user&.email == current_head[:email]`